### DISCUSSION TITLE
How to test a middleware properly?

### DISCUSSION THREAD

**Discussion by goorna** [nuxt/nuxt#18640] [Questions]
Hello everyone, I'm looking for advices about how to test middleware.
Even a simple and smaller one, I'm not sure about the way to do it, can you help me? I can't find documentation about this topic.

This one, for example:

```
export default defineNuxtRouteMiddleware(() => {
  const userStore = useUserStore();

  if (userStore.isLogged) {
    return navigateTo("/");
  }
});
```

Thank you in advance 🙏

**Comment by Baroshem** [comment]
Hey, have you seen this example page in the docs? 

https://nuxt.com/docs/examples/routing/middleware#middleware

It should be useful to test middleware out :)

**Comment by gpataciertocom** [comment]
I'm facing same problem. can't find anything related with middleware testing
